WIP: python312Packages.nicegui: update postPatch to fit to changed pyproject.toml

belamu opened the pull request:
The postPatch phase creates an error, documented in #383920 due to a change in the pyproject.toml upstream. This commit fits the replace string to the existing one.

## Things done
- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

belamu committed:
update postPatch to fit to changed pyproject.toml

The postPatch phase creates an error, documented in #383920 due to a change in the pyproject.toml upstream.
This commit fits the replace string to the existing one.

belamu commented:
@fabaff 

I won't have time in the next few days to check if that change works and is sufficiant to fix the build but you are welcome to use this first step.
Next week I would probably try it out.

belamu commented:
(Maybe this is a good time to stop and think about if the postPatch phase is still necessary. I have no clue.)

fabaff committed:
python312Packages.nicegui: 2.9.1 -> 2.11.1

Diff: https://github.com/zauberzeug/nicegui/compare/refs/tags/v2.9.1...v2.11.1

Changelog: https://github.com/zauberzeug/nicegui/releases/tag/v2.11.1

fabaff committed:
python312Packages.nicegui-highcharts: 2.0.2 -> 2.1.0

Diff: https://github.com/zauberzeug/nicegui-highcharts/compare/refs/tags/v2.0.2...v2.1.0

fabaff commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 383925`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python312Packages.nicegui</li>
    <li>python312Packages.nicegui-highcharts</li>
    <li>python312Packages.nicegui-highcharts.dist</li>
    <li>python312Packages.nicegui.dist</li>
    <li>python313Packages.nicegui</li>
    <li>python313Packages.nicegui-highcharts</li>
    <li>python313Packages.nicegui-highcharts.dist</li>
    <li>python313Packages.nicegui.dist</li>
  </ul>
</details>

belamu commented:
I was able to use this version in a nix-shell but unfortunately upon starting nicegui, I get `*.css` not found. The static files are missing.

To reproduce, use the python file `niceguiminimal.py`
```python
import nicegui
nicegui.ui.label("Hello World")
nicegui.ui.run()
```
the nix-shell
```nix
let
  pkgs = import (builtins.fetchGit
    url = "https://github.com/belamu/nixpkgs";
    ref = "patch-1";
  }) {};
in pkgs.mkShell {
    packages = with pkgs; [
      (python312.withPackages (python-pkgs: with python-pkgs; [
        nicegui
      ]))];}
```
and run `python niceguiminimal.py`. Then the output is
```
NiceGUI ready to go on http://localhost:8080, and http://172.20.30.202:8080
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/nicegui.css not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/quasar.prod.css not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/es-module-shims.js not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/socket.io.min.js not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/tailwindcss.min.js not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/fonts.css not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/nicegui.js not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/lang/en-US.umd.prod.js not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/vue.global.prod.js not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/quasar.umd.prod.js not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/tailwindcss.min.js not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/vue.global.prod.js not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/quasar.umd.prod.js not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/nicegui.js not found
http://127.0.0.1:8080/_nicegui/2.11.0.dev0/static/lang/en-US.umd.prod.js not found
```
And the web ui shows
` Connection lost. Trying to reconnect... `

belamu commented:
Indeed: in the publish workflow in. github/workflows, the version is replaced by the tag name:
```
  - name: set up Poetry
        uses: abatilo/actions-poetry@v4
      - name: get version
        run: echo "VERSION=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
      - name: Set version
        run: poetry version ${{ env.VERSION }}
```

belamu committed:
nicegui: patch version in pyproject.toml

this patching is also done in the github workflow for publishing
nicegui on pypy. Failing to patch it results in wrong directory names
for static files

belamu commented:
I've added a postPatch line that fixes the version in the `pyproject.toml` line. This actually changes the directory of the static files but unfortunately does not avoid the error.
Running the nix-shell mentioned above, I get the same error:
```
nix-shell shell-patched-minimal.nix 
this derivation will be built:
  /nix/store/hyqxpkxjnbcw1g6xwlvlgglldr6kq2hs-python3-3.12.8-env.drv
building '/nix/store/hyqxpkxjnbcw1g6xwlvlgglldr6kq2hs-python3-3.12.8-env.drv'...
created 384 symlinks in user environment

[nix-shell:~/Documents/Computer/Programmierprojekte/flinventory-gui]$ python
Python 3.12.8 (main, Dec  3 2024, 18:42:41) [GCC 14.2.1 20241116] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import nicegui
>>> nicegui.ui.run()
disabling auto-reloading because is is only supported when running from a file
NiceGUI ready to go on http://localhost:8080, and http://192.168.178.128:8080
http://127.0.0.1:8080/_nicegui/2.11.1/static/nicegui.css not found
http://127.0.0.1:8080/_nicegui/2.11.1/static/fonts.css not found
http://127.0.0.1:8080/_nicegui/2.11.1/static/quasar.prod.css not found
[........]
```

belamu commented:
The version update and removal of the postPatch is already done on master in MR #385669

belamu commented:
Changes not relevant anymore. Some part of the changes is tracked in #386199

The problems described above are currently still relevant and needed to understand #386199
